modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java

```diff
@@ -150,6 +224,26 @@ public void handleUpdateAll(
         });
     }
 
+    private void executeBatchGc(@Nullable HybridTimestamp newLwm) {
+        if (newLwm != null) {
+            @Nullable HybridTimestamp oldLwm;
+            do {
+                oldLwm = lastRecordedLwm.get();
+
+                if (oldLwm != null && newLwm.compareTo(oldLwm) <= 0) {
+                    break;
+                }
+            } while (!lastRecordedLwm.compareAndSet(oldLwm, newLwm));
```

Comment:
This method is called from RAFT listeners that serialize accesses. Do we really need a CAS here?

Reply:
Yes, raft commands are no longer execute serialized